enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,179 @@
+3. Administrator edits MicroShift config file to specify max file size and retention policy for logs
+4. Administrator selects action to take when storage limit or capacity reached
+5. MicroShift service is restarted to apply changes
+
+### API Extensions
+
+**Audit Log Policy Configuration:**
+
+- MicroShift config’s `apiServer` root-field must include a child node for storing log configurations.
+
+```yaml
+apiServer:
+  auditLog: MAP(STRING)INTERFACE
+```
+
+- The `apiServer.auditLog` field must include a child node for storing the audit log policy profile to use. Policies and descriptions are listed below. These are the same policies used by OpenShift.
+
+```yaml
+apiServer:
+  auditLog:
+    policy: STRING
+```
+
+| Policy             | Description                                                                                                                                                                                                                           |
+|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Default            | Logs only metadata for read and write requests; does not log request bodies except for OAuth access token requests. This is the default policy.                                                                                       |
+| WriteRequestBodies | In addition to logging metadata for all requests, logs request bodies for every write request to the API servers (create, update, patch, delete, deletecollection). This profile has more resource overhead than the Default profile. |
+| AllRequestBodiesIn | addition to logging metadata for all requests, logs request bodies for every read and write request to the API servers (get, list, create, update, patch). This profile has the most resource overhead.                               |
+| None               | No requests are logged; even OAuth access token requests and OAuth authorize token requests are not logged.                                                                                                                           |
+
+**Audit Log File Storage Location:**
+
+- `path` field to specify non-default audit log file paths.  Accepted values must be absolute paths.
+
+```yaml
+apiServer:
+  auditLog:
+    path: STRING
+```
+
+**Audit Log File Rotation:**
+
+- `maxFileSize` config variable to set maximum audit log file size. Accepted values must be a string with a number and unit suffix with no separator character, e.g. `10GB`. Valid units are "KB", "MB", "GB", "TB", "KiB", "MiB", "GiB", and "TiB".
+
+```yaml
+apiServer:
+  auditLog:
+    maxFileSize: STRING
+```
+
+**Audit Log Retention Policy:**
+
+- `retainPolicy` variable for system response when disk space full
+- Supported values: "ignore", "rotate", "terminate"
```

Comment:
Are any of these policies already supported by the audit logging code in the API server?

Reply:
No. This would be completely novel in OCP.  The existing api-servers don't seek to solve these situations.